man/man1/saver.1
.\" Automatically generated by Pandoc 2.19.2
.\"
.\" Define V font for inline verbatim, using C font in formats
.\" that render this, and otherwise B font.
.ie "\f[CB]x\f[]"x" \{\
. ftr V B
. ftr VI BI
. ftr VB B
. ftr VBI BI
.\}
.el \{\
. ftr V CR
. ftr VI CI
. ftr VB CB
. ftr VBI CBI
.\}
.TH "SAVER" "1" "December 06, 2021" "saver 4.0" "User Manual"
.hy
.SH NAME
.PP
\f[B]saver\f[R] - Convenience frontend for xscreensaver to manage
slideshows, screen saver folders, and xscreensaver commands
.SH SYNOPSIS
.PP
\f[B]saver\f[R] [ \f[B]-BDFHIPWX\f[R] path ] [ \f[B]-adflnprsxu\f[R] ] [
\f[B]-b\f[R] backgrounds_dir ] [ \f[B]-c\f[R] command ] [ \f[B]-t\f[R]
duration ]
.SH DESCRIPTION
.PP
Manage XScreenSaver slideshows, folders, and commands
.SH COMMAND LINE OPTIONS
.TP
\f[B]-b backgrounds_dir\f[R]
set the folder from which to get background images
.TP
\f[B]-c command\f[R]
\f[I]command\f[R] is one of \f[I]activate\f[R], \f[I]deactivate\f[R],
\f[I]blank\f[R], \f[I]demo\f[R], \f[I]exit\f[R], \f[I]lock\f[R],
\f[I]restart\f[R], \f[I]slides\f[R], or \f[I]start\f[R]
.TP
\f[B]-t duration\f[R]
specified in seconds
.TP
\f[B]-a\f[R]
indicates do not remove portrait format images
.TP
\f[B]-d\f[R]
indicates use /u/pictures/Domai rather than /u/pictures/Wallhaven
.TP
\f[B]-f\f[R]
indicates use /u/pictures/Femjoy rather than /u/pictures/Wallhaven
.TP
\f[B]-l\f[R]
indicates list the pre-created wallpaper subdirs available for the
selected type
.TP
\f[B]-n\f[R]
indicates tell me what you would do without doing anything
.TP
\f[B]-p\f[R]
indicates use /u/pictures/Playboy rather than /u/pictures/Wallhaven
.TP
\f[B]-r\f[R]
indicates restart xscreensaver prior to running command
.TP
\f[B]-s\f[R]
indicates use /u/pictures/Safe rather than /u/pictures/Wallhaven
.TP
\f[B]-x\f[R]
indicates use /u/pictures/X-Art rather than /u/pictures/Wallhaven
.TP
\f[B]-u\f[R]
displays usage message
.PP
The \f[B]-BFIPSWX path\f[R] options control the filesystem paths to
original image folders and pre-created folders of links to selected
image folders.
.TP
\f[B]-I path\f[R]
indicates use path as the top-level pathname to folders of original
images
.TP
\f[B]-D n1, -F n2, -S n3, -W n4, -X n5\f[R]
names of subfolders in the top-level folder
.TP
\f[B]-H path\f[R]
indicates use path as the full pathname to PICDIR where links will be
created
.TP
\f[B]-B name\f[R]
indicates use name as the subfolder in PICDIR where folders of links
live
.SH AUTHORS
.PP
Written by Ronald Record <<github@ronrecord.com>>
.SH LICENSING
.PP
SAVER is distributed under an Open Source license.
See the file LICENSE in the SAVER source distribution for information on
terms & conditions for accessing and otherwise using SAVER and for a
DISCLAIMER OF ALL WARRANTIES.
.SH BUGS
.PP
Submit bug reports online at:
<<https://gitlab.com/doctorfree/DoctorFreeScripts/issues>>
.SH SEE ALSO
.PP
Full documentation and sources at:
<<https://gitlab.com/doctorfree/DoctorFreeScripts>>


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

Jan 27, 2022 Version 4.0.1 Release 2
    Minor release update with improved support for SD management utilities,
    extending supported platforms to Linux for the SD utils.

Jan 17, 2022 Version 4.0.1 Release 1
    Release update to Version 4.0.1 including additional man pages
    and Zip format distribution archive. Minor bug fixes and improvements.
    RPM installation fixes.

Nov 26, 2021 Version 4.0 Release 1
    This release adds Debian and RPM format installation package support with
    improved support for ease of installation, configuration, update, and removal.

Wed Jan 26 14:27:43 2022 -0800 b752940 :
   Exec /bin/bash in systemd service to comply with Fedora restrictions
Tue Jan 25 08:37:12 2022 -0800 f38080d :
   Add journalclean command
Tue Jan 25 07:14:52 2022 -0800 d4b92fc :
   Set backgrounds folder to /media/user/Transcend by default on Linux
Mon Jan 24 18:21:57 2022 -0800 363a940 :
   Change default SD device to /dev/sdd
Mon Jan 24 10:17:35 2022 -0800 3be3fab :
   Add documentation for sdmount and sdumount
Sun Jan 23 19:48:26 2022 -0800 4088f8d :
   Add sdmount and sdumount commands
Sun Jan 23 10:31:20 2022 -0800 68b537d :
   Doc for sdlist
Sun Jan 23 10:04:48 2022 -0800 04f8708 :
   Add sdlist command
Sat Jan 22 16:03:14 2022 -0800 02729e7 :
   Extend SD utilities with Linux support
Fri Jan 21 07:50:47 2022 -0800 6be5d58 :
   Muriel aliases
Thu Jan 20 11:45:13 2022 -0800 24d045d :
   Redirect stderr and stdout to parent process during RPM scriptlets
Tue Jan 18 17:17:31 2022 -0800 2152b2d :
   Use DISPLAY env var if set
Tue Jan 18 13:20:58 2022 -0800 85da437 :
   Add doctorinfo script
Mon Jan 17 13:52:51 2022 -0800 65cd49e :
   Adopt release creation during Gitlab CI
Mon Jan 17 13:42:38 2022 -0800 735ff4e :
   Do not invoke dl_fred_scripts during postinstall, new version string
Mon Jan 17 12:06:57 2022 -0800 6bc24c5 :
   Move pvoutput to MirrorCommand
Mon Jan 17 12:06:31 2022 -0800 8dd49ed :
   Bump release
Fri Jan 14 13:52:28 2022 -0800 1ab3135 :
   Update miner scripts
Fri Jan 14 10:49:45 2022 -0800 988389c :
   Add disable-update-notifier and pkg/tools/get_release
Mon Jan 10 12:23:00 2022 -0800 6e4799a :
   Do not package or export gdrive utils
Mon Jan 10 12:19:41 2022 -0800 9c85af0 :
   Remove gdrive files as they are now packaged in their own repo
Sun Jan 9 17:43:49 2022 -0800 8962417 :
   disable rpmbuild strip
Sun Jan 9 17:18:32 2022 -0800 5af1d70 :
   Fix files section of spec file
Sun Jan 9 17:10:20 2022 -0800 905d402 :
   Move all files into /usr/local
Sun Jan 9 16:39:42 2022 -0800 e27d740 :
   Try noninteractive rpm install again
Sun Jan 9 15:35:04 2022 -0800 64ddc3a :
   Fix for CI
Sun Jan 9 15:28:21 2022 -0800 fd5915b :
   export noninteractive
Sun Jan 9 15:18:00 2022 -0800 c6e51e9 :
   Copy rpm dist tree in install phase
Sun Jan 9 14:46:58 2022 -0800 bfa4e07 :
   Install rpm in CI in noninteractive mode with TZ set
Sun Jan 9 14:17:31 2022 -0800 53255f8 :
   Use ubuntu:latest CI image
Sun Jan 9 14:09:05 2022 -0800 c6963e9 :
   Version 4.0 release 3
Sun Jan 9 14:04:52 2022 -0800 fd33ea1 :
   Install zip if not found, create zip artifacts in CI
Thu Jan 6 11:57:39 2022 -0800 3b3f004 :
   Add drive to projects
Mon Jan 3 20:15:25 2022 -0800 748b3e0 :
   Add author attribution
Mon Jan 3 16:22:13 2022 -0800 236f864 :
   Add -l option to list sync folders
Mon Jan 3 16:11:26 2022 -0800 d30988a :
   Renamed getfileids as getfolderids
Mon Jan 3 13:01:47 2022 -0800 97d0e8d :
   Check for gdrive binary in PATH
Mon Jan 3 12:38:50 2022 -0800 ded8f3d :
   Add options to getfolderids, start on support for spaces in file and folder names
Mon Jan 3 10:24:36 2022 -0800 627a90d :
   Support spaces in folder names, retrieve all folder ids
Sun Jan 2 21:10:06 2022 -0800 1ec7cb9 :
   Export gdrive utils
Sun Jan 2 21:07:06 2022 -0800 40e2bdb :
   Add gdrive scripts to exports
Sun Jan 2 21:04:30 2022 -0800 e9d2a92 :
   Add comment for developer's guide to api
Sun Jan 2 20:19:56 2022 -0800 c674bc0 :
   Add sync_from_drive, update sync2drive
Sun Jan 2 11:48:46 2022 -0800 285acd8 :
   Updated gdrive utility scripts
Sat Jan 1 20:14:40 2022 -0800 a670a0d :
   Add new scripts for command line management of Google Drive
Fri Dec 31 08:52:06 2021 -0800 a1659d2 :
   Add Mirco Cabbia alias
Tue Dec 28 09:50:38 2021 -0800 904ab6d :
   Rename MirrorCommandLine to MirrorCommand
Mon Dec 27 18:20:34 2021 -0800 3554d70 :
   New model alias
Mon Dec 27 08:43:35 2021 -0800 5a2a524 :
   New model alias
Mon Dec 27 08:23:33 2021 -0800 a981875 :
   Create man page for md2man
Sun Dec 26 17:48:24 2021 -0800 6d83be7 :
   Set output dir based on man page section in md2man
Sat Dec 25 08:04:35 2021 -0800 ef79abb :
   Added Colombian wallpapers
Fri Dec 24 11:34:02 2021 -0800 0c9e9cc :
   Adjusted aliases
Thu Dec 23 09:38:25 2021 -0800 c2b5191 :
   Updated model name, add Mila Azul to Ukrainians
Wed Dec 22 16:31:35 2021 -0800 d4296b4 :
   Use HEAD to determine branch
Tue Dec 21 14:01:20 2021 -0800 526124a :
   Bump release number, update changelog
Tue Dec 21 10:26:02 2021 -0800 8c3a636 :
   Add all my clone and build scripts
Tue Dec 21 09:43:04 2021 -0800 d625a49 :
   Check and set location of Synergy client binary and system tools in Synergy startup control script
Tue Dec 21 09:14:43 2021 -0800 7a101e1 :
   Add man page for sdwrite, add SEE ALSO pages to all SD card commands
Tue Dec 21 08:40:29 2021 -0800 58b2dfc :
   Move sdwrite from MirrorCommandLine package to here
Sat Dec 18 17:18:51 2021 -0800 148f60a :
   Set synergyctl IP in postinstall
Sat Dec 18 16:39:57 2021 -0800 a8df6ce :
   Add synergyctl script to control Synergy start/stop
Fri Dec 17 07:15:45 2021 -0800 30fd5e5 :
   New model aliases
Mon Dec 13 21:11:37 2021 -0800 2c51ae3 :
   Fix typo in rmbrokenpkg
Mon Dec 13 20:18:02 2021 -0800 e5436b5 :
   Add rmbrokenpkg
Sun Dec 12 13:40:26 2021 -0800 2a4714d :
   Add note about Windows System for Linux

Wallpapers/saver.sh
#!/bin/bash
#
## @file Wallpapers/saver.sh
## @brief Convenience frontend for xscreensaver to manage slideshow screen saver folders
## @author Ronald Joe Record (rr at ronrecord dot com)
## @copyright Copyright (c) 2018, Ronald Joe Record, all rights reserved.
## @date Written 17-Oct-2018
## @version 1.0.5
##
# --------------------------------------------------------------------------------------
# Modify the following to reflect the location and names of image folders on your system
# (Alternately, these settings can be specified on the command line)
# --------------------------------------------------------------------------------------
# These two define the location of pre-created folders of links to background images
PICDIR="$HOME/Pictures"
BGNDS="Backgrounds"

# The xscreensaver configuration file to be edited and the command to look for in it
CONF="$HOME/.xscreensaver"
SHOW="glslideshow -root -duration"

# The top of the directory hierarchy that contains the original image folders
ITOP="/u/pictures"
# The names of those original image folders
DBACK="Domai"
FBACK="Femjoy"
PBACK="Playboy"
SBACK="Safe"
WBACK="Wallhaven"
XBACK="X-Art"
# ----------------------------
# End of user-defined settings
# ----------------------------

# Default paths and folder names for original images and pre-created folders of links
WHVN="$ITOP/$WBACK"
SAFE="$ITOP/$SBACK"
DMAI="$ITOP/$DBACK"
FMJY="$ITOP/$FBACK"
PLBY="$ITOP/$PBACK"
XART="$ITOP/$XBACK"
WDIR="$WHVN"
WBCK="$WBACK"
BACK="$PICDIR/$BGNDS/$WBCK"

current() {
    echo ""
    echo "Current XScreenSaver image directory set to $BGDIR"
    echo "Current XScreenSaver glslideshow duration set to $ODUR"
    echo "Current XScreenSaver mode set to $MODE"
}

usage() {
    echo "Usage: saver [-BDFHIPWX path] [-adflnprsxu] [-b backgrounds_dir] [-c command] [-t duration]"
    echo ""
    echo "Where <backgrounds dir> can be one of:"
    echo -e "\tAny subdir in $WHVN"
    echo -e "\tAny subdir in $DMAI if -d has been specified"
    echo -e "\tAny subdir in $FMJY if -f has been specified"
    echo -e "\tAny subdir in $PLBY if -p has been specified"
    echo -e "\tAny subdir in $SAFE if -s has been specified"
    echo -e "\tAny subdir in $XART if -x has been specified"
    echo -e "\tOne of the pre-created dirs in $BACK"
    echo -e "\n\tCurrent pre-created dirs include"
    echo -e "\n\tModels:"
    echo -e "\t[Alisa|Carisha|Corinna|Jasmine_A|Li_Moon|Mila]"
    echo -e "\n\tPhotographers/Artists/Sites:"
    echo -e "\t[Femjoy|Nikolaev|Russians|Safin|Sakimichan|Soell|Tuigirl]"
    echo -e "\n\tGeneral:"
    echo -e "\t[All|Anime|Favs|Fractals|Hentai|Owls|Waterfalls]"
    echo -e "\n\tSafe:"
    echo -e "\t[All|Favs|Dragonflies|Fractals|Owls|Space|Waterfalls]"
    echo ""
    echo "<command> can be one of:"
    echo "    [activate|deactivate|blank|demo|exit|lock|restart|slides|start]"
    echo ""
    echo "<duration> is specified in seconds"
    echo ""
    echo "-a indicates do not remove portrait format images"
    echo "-d indicates use $DMAI rather than $WHVN"
    echo "-f indicates use $FMJY rather than $WHVN"
    echo "-l indicates list the pre-created wallpaper subdirs available for the selected type"
    echo "-n indicates tell me what you would do without doing anything"
    echo "-p indicates use $PLBY rather than $WHVN"
    echo "-r indicates restart xscreensaver prior to running command"
    echo "-s indicates use $SAFE rather than $WHVN"
    echo "-x indicates use $XART rather than $WHVN"
    echo "-u displays this usage message"
    echo ""
    echo "The '-BFIPSWX path' options control the filesystem paths to original image folders"
    echo "and pre-created folders of links to selected image folders."
    echo ""
    echo "-I path indicates use path as the top-level pathname to folders of original images"
    echo "-D n1,-F n2,-S n3,-W n4, and -X n5 are the names of subfolders in the top-level folder"
    echo "-H path indicates use path as the full pathname to PICDIR where links will be created"
    echo "-B name indicates use name as the subfolder in PICDIR where folders of links live"
    echo ""
    echo "Current settings for these user defined paths and folder names are:"
    echo ""
    echo "Top-level original image folder ITOP = $ITOP"
    echo "-D n1 specified \$ITOP/\$n1 = $ITOP/$DBACK"
    echo "-F n2 specified \$ITOP/\$n2 = $ITOP/$FBACK"
    echo "-P n3 specified \$ITOP/\$n3 = $ITOP/$PBACK"
    echo "-S n4 specified \$ITOP/\$n4 = $ITOP/$SBACK"
    echo "-W n5 specified \$ITOP/\$n5 = $ITOP/$WBACK"
    echo "-X n6 specified \$ITOP/\$n6 = $ITOP/$XBACK"
    echo "-H path, -B name specified path/name/back = $PICDIR/$BGNDS/$WBCK"
    if [ "$LIST" ]
    then
        listem
    else
        current
    fi
    exit 1
}

linkem() {
    case $WDIR in
        $WHVN|$SAFE|$DMAI|$PLBY)
            for pic in "$1"/*
            do
                [ "$pic" == "$1/*" ] && {
                    echo "No images found for $1 in $WDIR"
                    continue
                }
                ln -s "$pic" .
            done
            ;;
        *)
            for pic in "$1"/*/*
            do
                [ "$pic" == "$1/*/*" ] && {
                    echo "No images found for $1 in $WDIR"
                    continue
                }
                ln -s "$pic" .
            done
            ;;
    esac
    rm -f *.txt
    rm -f *.gif

    [ "$rmportrait" ] && {
        inst=`type -p identify`
        [ "$inst" ] && {
            for i in *.jpg *.jpeg *.png
            do
                [ "$i" == "*.jpg" ] && continue
                [ "$i" == "*.jpeg" ] && continue
                [ "$i" == "*.png" ] && continue
                GEO=`identify "$i" | awk ' { print $(NF-6) } '`
                W=`echo $GEO | awk -F "x" ' { print $1 } '`
                # Remove if width not greater than 1000
                [ "$W" ] && [ $W -gt 1000 ] || {
                    rm -f "$i"
                    continue
                }
                H=`echo $GEO | awk -F "x" ' { print $2 } '`
                # Remove if height not greater than 750
                [ "$H" ] && [ $H -gt 750 ] || {
                    rm -f "$i"
                    continue
                }
                # Remove if width not greater than height
                [ "$W" ] && [ "$H" ] && [ $W -gt $H ] || rm -f "$i"
            done
        }
    }
}

listem() {
    echo -e "\nPre-Created Wallpaper folders available for $WBCK :\n"
    ls --color=auto $BACK
    current
}

mkbgdir() {
    [ -d $BACK ] || {
        echo "$BACK does not exist or is not a directory. Exiting."
        exit 1
    }
    cd $BACK
    sub="$1"
    [ -d $sub ] || {
        mkdir $sub
        cd $sub
        if [ -d $WDIR/$sub ]
        then
            linkem $WDIR/$sub
        else
            if [ -d $WDIR/Models/$sub ]
            then
                linkem $WDIR/Models/$sub
            else
                if [ -d $WDIR/Photographers/$sub ]
                then
                    linkem $WDIR/Photographers/$sub
                else
                    echo "No $WDIR subfolder found for $sub. Exiting"
                    exit 1
                fi
            fi
        fi
        cd $HERE
    }
}

HERE=`pwd`
BGDIR=`grep imageDirectory: $CONF | awk ' { print $2 } '`
ODUR=`grep glslideshow $CONF | awk -F ":" ' { print $2 } ' | awk ' { print $4 } '`
MODE=`grep mode: $CONF | awk ' { print $2 } '`

# No arguments we take to mean activate the screensaver
[ "$1" ] || {
    xscreensaver-command -activate > /dev/null
    exit 0
}

BDIR=
COMM=
SECS=
LIST=
TELL=
REST=
USAGE=
dir_arg=
com_arg=
dom_arg=
fem_arg=
pby_arg=
art_arg=
saf_arg=
switch=0
rmportrait=1

while getopts B:D:F:H:I:P:W:X:b:c:t:adflnprsxu flag; do
    case $flag in
        B)
            BGNDS="$OPTARG"
            ;;
        I)
            ITOP="$OPTARG"
            ;;
        D)
            DBACK="$OPTARG"
            ;;
        F)
            FBACK="$OPTARG"
            ;;
        H)
            PICDIR="$OPTARG"
            ;;
        P)
            PBACK="$OPTARG"
            ;;
        S)
            SBACK="$OPTARG"
            ;;
        W)
            WBACK="$OPTARG"
            ;;
        X)
            XBACK="$OPTARG"
            ;;
        a)
            rmportrait=
            ;;
        b)
            dir_arg="$OPTARG"
            ;;
        c)
            com_arg="$OPTARG"
            ;;
        d)
            WDIR="$DMAI"
            WBCK="$DBACK"
            BACK="$PICDIR/$BGNDS/$WBCK"
            dom_arg=1
            switch=`expr $switch + 1`
            ;;
        f)
            WDIR="$FMJY"
            WBCK="$FBACK"
            BACK="$PICDIR/$BGNDS/$WBCK"
            fem_arg=1
            switch=`expr $switch + 1`
            ;;
        l)
            LIST=1
            ;;
        n)
            TELL=1
            ;;
        p)
            WDIR="$PLBY"
            WBCK="$PBACK"
            BACK="$PICDIR/$BGNDS/$WBCK"
            pby_arg=1
            switch=`expr $switch + 1`
            ;;
        r)
            REST=1
            ;;
        s)
            WDIR="$SAFE"
            WBCK="$SBACK"
            BACK="$PICDIR/$BGNDS/$WBCK"
            saf_arg=1
            switch=`expr $switch + 1`
            ;;
        t)
            SECS="$OPTARG"
            ;;
        x)
            WDIR="$XART"
            WBCK="$XBACK"
            BACK="$PICDIR/$BGNDS/$WBCK"
            art_arg=1
            switch=`expr $switch + 1`
            ;;
        u)
            USAGE=1
            ;;
    esac
done
shift $(( OPTIND - 1 ))

# Only one of -f, -s, and -x can be specified
[ $switch -gt 1 ] && {
    echo "Command line options -d,-f,-p,-s, and -x are mutually exclusive. Use only one or none."
    echo "Exiting."
    exit 1
}

# Reset these just in case they were specified on the command line
WHVN="$ITOP/$WBACK"
DMAI="$ITOP/$DBACK"
FMJY="$ITOP/$FBACK"
PLBY="$ITOP/$PBACK"
SAFE="$ITOP/$SBACK"
XART="$ITOP/$XBACK"
if [ "$fem_arg" ]
then
    WDIR="$FMJY"
    WBCK="$FBACK"
else
    if [ "$art_arg" ]
    then
        WDIR="$XART"
        WBCK="$XBACK"
    else
        if [ "$saf_arg" ]
        then
            WDIR="$SAFE"
            WBCK="$SBACK"
        else
            if [ "$dom_arg" ]
            then
                WDIR="$DMAI"
                WBCK="$DBACK"
            else
                if [ "$pby_arg" ]
                then
                    WDIR="$PLBY"
                    WBCK="$PBACK"
                else
                    WDIR="$WHVN"
                    WBCK="$WBACK"
                fi
            fi
        fi
    fi
fi
BACK="$PICDIR/$BGNDS/$WBCK"

[ "$USAGE" ] && usage
[ "$LIST" ] && listem

[ "$dir_arg" ] && {
    case "$dir_arg" in
        alisa|alisa_i|Alisa_I|Alisa) BDIR="Alisa_I"
               ;;
        all|All) BDIR="All"
               ;;
        anime|Anime) BDIR="Anime"
               ;;
        carisha|Carisha) BDIR="Carisha"
               ;;
        corinna|Corinna) BDIR="Corinna"
               ;;
        favs|Favs) BDIR="Favs"
               ;;
        femjoy|Femjoy) BDIR="All"
               ;;
        fractals|Fractals) BDIR="Fractals"
               ;;
        hentai|Hentai) BDIR="Hentai"
               ;;
        jasmine*|Jasmine*) BDIR="Jasmine_A"
               ;;
        li_moon|Li_Moon|li|Li) BDIR="Li_Moon"
               ;;
        mila_a|Mila_A) BDIR="Mila_A"
               ;;
        mila_k|Mila_K) BDIR="Mila_K"
               ;;
        milla|Milla) BDIR="Milla"
               ;;
        mila|Mila)
               if [ "$WBCK" == "$XBACK" ]
               then
                   BDIR="Mila_K"
               else
                   BDIR="Mila_A"
               fi
               ;;
        natalia_andreeva|Natalia_Andreeva) BDIR="Natalia_Andreeva"
               ;;
        owls|Owls) BDIR="Owls"
               ;;
        russian*|Russian*) BDIR="Russian_women"
               ;;
        safe|Safe) BDIR="Fractals"
               ;;
        safin|Safin|Marat*|marat*) BDIR="Marat_Safin"
               ;;
        saki|sakimichan|Sakimichan|Saki) BDIR="Sakimichan"
               ;;
        soell|Soell) BDIR="Stefan_Soell"
               ;;
        sybil*|Sybil*)
               if [ "$WBCK" == "$XBACK" ]
               then
                   BDIR="Sybil"
               else
                   BDIR="Sybil_A"
               fi
               ;;
        tui|tuigirl|Tui|Tuigirl) BDIR="Tuigirl"
               ;;
        vlad|vladimir|Vlad|Vladimir|*Nikolaev) BDIR="Vladimir_Nikolaev"
               ;;
        waterfalls|Waterfalls) BDIR="Waterfalls"
               ;;
        xart|Xart|x-art|X-Art) BDIR="All"
               ;;
        *) BDIR="$dir_arg"
               ;;
    esac
}

[ "$com_arg" ] && {
    case "$com_arg" in
        activate|deactivate|demo|exit|lock|restart) COMM="$com_arg"
            ;;
        blank|noslides)
            if [ "$TELL" ]
            then
                echo "cat $CONF | sed -e s/mode:\t\tone/mode:\t\tblank/ > /tmp/xscr$$"
                echo "cp /tmp/xscr$$ $CONF"
                echo "rm -f /tmp/xscr$$"
            else
                cat $CONF | sed -e "s/mode:\t\tone/mode:\t\tblank/" > /tmp/xscr$$
                cp /tmp/xscr$$ $CONF
                rm -f /tmp/xscr$$
            fi
            ;;
        one|noblank|slides)
            if [ "$TELL" ]
            then
                echo "cat $CONF | sed -e s/mode:\t\tblank/mode:\t\tone/ > /tmp/xscr$$"
                echo "cp /tmp/xscr$$ $CONF"
                echo "rm -f /tmp/xscr$$"
            else
                cat $CONF | sed -e "s/mode:\t\tblank/mode:\t\tone/" > /tmp/xscr$$
                cp /tmp/xscr$$ $CONF
                rm -f /tmp/xscr$$
            fi
            ;;
        start)
            if [ "$TELL" ]
            then
                echo "xscreensaver &"
            else
                xscreensaver &
            fi
            ;;
        *)  echo "Unrecognized xscreensaver-command $com_arg - exiting"
            exit 1
            ;;
    esac
}

[ "$BDIR" ] && {
  [ -d "$PICDIR/$BGNDS/$WBCK/$BDIR" ] || mkbgdir $BDIR
  [ "$PICDIR/$BGNDS/$WBCK/$BDIR" == "$BGDIR" ] || {
    ODIR=`basename $BGDIR`
    T_OWBK=`dirname $BGDIR`
    OWBK=`basename $T_OWBK`
    if [ "$TELL" ]
    then
        echo "cat $CONF | sed -e s/$BGNDS\/$OWBK\/$ODIR/$BGNDS\/$WBCK\/$BDIR/ > /tmp/xscr$$"
        echo "cp /tmp/xscr$$ $CONF"
        echo "rm -f /tmp/xscr$$"
    else
        cat $CONF | sed -e "s/$BGNDS\/$OWBK\/$ODIR/$BGNDS\/$WBCK\/$BDIR/" > /tmp/xscr$$
        cp /tmp/xscr$$ $CONF
        rm -f /tmp/xscr$$
    fi
  }
}
[ "$SECS" ] && {
  [ $ODUR -eq $SECS ] || {
    if [ "$TELL" ]
    then
        echo "cat $CONF | sed -e s/$SHOW $ODUR -pan $ODUR/$SHOW $SECS -pan $SECS/ > /tmp/xscr$$"
        echo "cp /tmp/xscr$$ $CONF"
        echo "rm -f /tmp/xscr$$"
    else
        cat $CONF | sed -e "s/$SHOW $ODUR -pan $ODUR/$SHOW $SECS -pan $SECS/" > /tmp/xscr$$
        cp /tmp/xscr$$ $CONF
        rm -f /tmp/xscr$$
    fi
  }
}

[ "$COMM" ] && {
    if [ "$TELL" ]
    then
        [ "$REST" ] && {
          echo -e "\nxscreensaver-command -exit > /dev/null"
          echo "xscreensaver &"
        }
        echo -e "\nxscreensaver-command -$COMM > /dev/null"
    else
        [ "$REST" ] && {
          xscreensaver-command -exit > /dev/null
          xscreensaver &
        }
        xscreensaver-command -$COMM > /dev/null
    fi
}
